tests/restapi/test_identifiers.py

```diff
@@ -8,62 +8,206 @@
+    current_namespace = current_namespace['subspaces'][0]
+    query_all = orm.QueryBuilder().append(orm.Node, filters=get_full_type_filters(current_namespace['full_type']))
+    assert len(current_namespace['subspaces']) == 1
+    assert query_all.count() == 3
+
+    # One branch for each registered plugin and one for all unregistered
+    # (there are only unregistered here)
+    current_namespace = current_namespace['subspaces'][0]
+    query_all = orm.QueryBuilder().append(orm.Node, filters=get_full_type_filters(current_namespace['full_type']))
+    assert len(current_namespace['subspaces']) == 1
+    assert query_all.count() == 3
+
+    # There are only two  process types: unregistered_type1 (1) and unregistered_type2 (2)
+    # The unregistered branch does not currently return a queryable full_type
+    current_namespace = current_namespace['subspaces'][0]
+    assert len(current_namespace['subspaces']) == 2
+
+    type_namespace = current_namespace['subspaces'][0]
+    query_type = orm.QueryBuilder().append(orm.Node, filters=get_full_type_filters(type_namespace['full_type']))
+    assert len(type_namespace['subspaces']) == 1
+    assert query_type.count() == 1
+
+    type_namespace = current_namespace['subspaces'][1]
+    query_type = orm.QueryBuilder().append(orm.Node, filters=get_full_type_filters(type_namespace['full_type']))
+    assert len(type_namespace['subspaces']) == 2
+    assert query_type.count() == 2
+
+    # Finally we check each specific subtype (1 for unregistered_type1 and 2 for unregistered_type2)
+    # These are lead nodes without any further subspace
+    type_namespace = current_namespace['subspaces'][0]['subspaces'][0]
+    query_type = orm.QueryBuilder().append(orm.Node, filters=get_full_type_filters(type_namespace['full_type']))
+    assert len(type_namespace['subspaces']) == 0
+    assert query_type.count() == 1
+
+    type_namespace = current_namespace['subspaces'][1]['subspaces'][0]
+    query_type = orm.QueryBuilder().append(orm.Node, filters=get_full_type_filters(type_namespace['full_type']))
+    assert len(type_namespace['subspaces']) == 0
+    assert query_type.count() == 1
+
+    type_namespace = current_namespace['subspaces'][1]['subspaces'][1]
+    query_type = orm.QueryBuilder().append(orm.Node, filters=get_full_type_filters(type_namespace['full_type']))
+    assert len(type_namespace['subspaces']) == 0
+    assert query_type.count() == 1
+
+
+@pytest.mark.parametrize('NodeClass', [orm.CalcFunctionNode, orm.Dict])
+@pytest.mark.usefixtures('clear_database_before_test')
+def test_full_type_backwards_compatibility(NodeClass, restapi_server, server_url):
+    """Test the compatibility with old process_type entries."""
+    # pylint: disable=invalid-name
+
+    node_empty = NodeClass()
+    node_empty.process_type = ''
+    node_empty.store()
+
+    node_nones = NodeClass()
+    node_nones.process_type = None
+    node_nones.store()
```

Comment:
should we also add this case to `test_get_full_type_filters`, i.e. checking there the correct output of `get_full_type_filters` for the anomaly process types?

Reply:
What do you mean checking the "correct output" of `get_full_type_filters`? In this test below I'm using `get_full_type_filters` to get the filters for the queries I'm using to check, thus checkign the "correctedness" in the sense that it gives filters that work on the querybuilder. Do you mean checking them literally instead of functionally? Isn't this more versatile?

Reply:
in `test_get_full_type_filters` you do not add test cases for `process_type in ['', None]`, but the function contains explicit logic branches for it. Now, you *indirectly* test it through these tests here below, but that is in a roundabout way and not very direct. It should be tested as closely to the code itself as possible as to make it very clear

Reply:
Ok, I think I understand what you mean. I will add some more explicit tests there.